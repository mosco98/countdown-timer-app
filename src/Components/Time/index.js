import '../../Assets/css/style.css'

import classNames from 'classnames'
import React from 'react'
import { Pause, Play } from 'react-feather'

const Time = ({
  timePaused,
  startTimeHandler,
  timeStarted,
  minutes,
  seconds,
  pauseTimeHandler,
  timeRed,
  timeBlink
}) => {
  return (
    <div className="d-flex align-items-center justify-content-center">
      <div className="container w-100 d-flex align-items-center justify-content-around time-ctn">
        <h1
          className={classNames('time ', {
            'time red-text': timeRed,
            'time blinking': timeBlink
          })}>
          {minutes < 10 ? '0' + minutes : minutes}:{seconds < 10 ? '0' + seconds : seconds}
        </h1>
        {timeStarted ? (
          <div className="d-flex align-items-center justify-content-center ml-2 mt-auto mb-auto">
            {timePaused ? (
              <Play
                className={classNames({
                  'red-text': timeRed
                })}
                size={45}
                style={{ cursor: 'pointer' }}
                onClick={() => startTimeHandler('resume')}
                title="Resume"
              />
            ) : (
              <Pause
                className={classNames({
                  'red-text': timeRed
                })}
                size={45}
                style={{ cursor: 'pointer' }}
                onClick={pauseTimeHandler}
                title="Pause"
              />
            )}
          </div>
        ) : null}
      </div>
    </div>
  )
}

export default Time
