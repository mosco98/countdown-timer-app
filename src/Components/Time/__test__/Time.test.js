import { fireEvent, render } from '@testing-library/react'
import React from 'react'
import renderer from 'react-test-renderer'

import Time from '..'

describe('Time component initial rendering', () => {
  it('renders correctly', () => {
    render(<Time />)
  })
})

describe('Rendering pause and resume time button', () => {
  it('rendering pause button', () => {
    const props = {
      timeStarted: true,
      timePaused: false
    }
    const { timeStarted, timePaused } = props
    const component = renderer.create(<Time timeStarted={timeStarted} timePaused={timePaused} />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('rendering resume button', () => {
    const props = {
      timeStarted: true,
      timePaused: true
    }
    const { timeStarted, timePaused } = props
    const component = renderer.create(<Time timeStarted={timeStarted} timePaused={timePaused} />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})

describe('Button functionality', () => {
  it('pause button triggers the pauseTimerHandler function when clicked', () => {
    const pauseTimeHandler = jest.fn()
    const props = {
      timeStarted: true,
      timePaused: false
    }
    const { timeStarted, timePaused } = props
    const { queryByTestId } = render(
      <Time pauseTimeHandler={pauseTimeHandler} timeStarted={timeStarted} timePaused={timePaused} />
    )
    fireEvent.click(queryByTestId('pause-btn'))
    expect(pauseTimeHandler).toHaveBeenCalled()
  })
  it('resume button triggers the startTimerHandler function when clicked', () => {
    const startTimeHandler = jest.fn()
    const props = {
      timeStarted: true,
      timePaused: true
    }
    const { timeStarted, timePaused } = props
    const { queryByTestId } = render(
      <Time startTimeHandler={() => startTimeHandler('resume')} timeStarted={timeStarted} timePaused={timePaused} />
    )
    fireEvent.click(queryByTestId('resume-btn'))
    expect(startTimeHandler).toHaveBeenCalled()
  })
})
