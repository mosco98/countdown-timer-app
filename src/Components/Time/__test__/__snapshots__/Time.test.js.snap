// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering pause and resume time button rendering pause button 1`] = `
<div
  className="d-flex align-items-center justify-content-center"
>
  <div
    className="container w-100 d-flex align-items-center justify-content-around time-ctn"
  >
    <h1
      className="time ml-2"
    >
      :
    </h1>
    <div
      className="d-flex align-items-center justify-content-center ml-2 mt-auto mb-auto"
    >
      <svg
        className=""
        data-testid="pause-btn"
        fill="none"
        height={45}
        stroke="currentColor"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="2"
        style={
          Object {
            "cursor": "pointer",
          }
        }
        title="Pause"
        viewBox="0 0 24 24"
        width={45}
        xmlns="http://www.w3.org/2000/svg"
      >
        <rect
          height="16"
          width="4"
          x="6"
          y="4"
        />
        <rect
          height="16"
          width="4"
          x="14"
          y="4"
        />
      </svg>
    </div>
  </div>
</div>
`;

exports[`Rendering pause and resume time button rendering resume button 1`] = `
<div
  className="d-flex align-items-center justify-content-center"
>
  <div
    className="container w-100 d-flex align-items-center justify-content-around time-ctn"
  >
    <h1
      className="time ml-2"
    >
      :
    </h1>
    <div
      className="d-flex align-items-center justify-content-center ml-2 mt-auto mb-auto"
    >
      <svg
        className=""
        data-testid="resume-btn"
        fill="none"
        height={45}
        onClick={[Function]}
        stroke="currentColor"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="2"
        style={
          Object {
            "cursor": "pointer",
          }
        }
        title="Resume"
        viewBox="0 0 24 24"
        width={45}
        xmlns="http://www.w3.org/2000/svg"
      >
        <polygon
          points="5 3 19 12 5 21 5 3"
        />
      </svg>
    </div>
  </div>
</div>
`;
